File: models/Gpt-3.5-turbo/py/code/additional_problems/Advertisement.py

n = int(input())
heights = list(map(int, input().split()))

left = 0
right = n - 1
max_area = 0

while left < right:
    width = right - left
    height = min(heights[left], heights[right])
    area = width * height
    max_area = max(max_area, area)

    if heights[left] < heights[right]:
        left += 1
    else:
        right -= 1

print(max_area)

File: models/Gpt-3.5-turbo/cpp/code/problems/additional_problems/Advertisement.cpp

#include <iostream>
#include <stack>
#include <vector>

using namespace std;

int main() {
    int n;
    cin >> n;

    vector<int> heights(n);
    for (int i = 0; i < n; i++) {
        cin >> heights[i];
    }

    stack<int> st;
    int maxArea = 0;

    for (int i = 0; i < n; i++) {
        while (!st.empty() && heights[st.top()] >= heights[i]) {
            int height = heights[st.top()];
            st.pop();

            int width = st.empty() ? i : (i - st.top() - 1);
            maxArea = max(maxArea, height * width);
        }
        st.push(i);
    }

    while (!st.empty()) {
        int height = heights[st.top()];
        st.pop();

        int width = st.empty() ? n : (n - st.top() - 1);
        maxArea = max(maxArea, height * width);
    }

    cout << maxArea << endl;

    return 0;
}
